common/component/kafka/auth.go

```diff
@@ -128,6 +131,13 @@ func (m *mskAccessTokenProvider) Token() (*sarama.AccessToken, error) {
 			}, nil
 		}))
 		return &sarama.AccessToken{Token: token}, err
+	} else if m.awsIamRoleArn != "" {
+		// Using a different context since the context from init may have been cancelled when token is refreshed
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
Context implementation will be updated based on the outcome of https://github.com/dapr/components-contrib/pull/3305